internal/controller/controllers/hypershiftserviceconfig_controller_test.go

```diff
@@ -0,0 +1,83 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/assisted-service/internal/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	kubeconfigSecret = "secret"
+)
+
+func newHypershiftServiceConfigRequest(asc *aiv1beta1.HypershiftServiceConfig) ctrl.Request {
+	namespacedName := types.NamespacedName{
+		Namespace: asc.ObjectMeta.Namespace,
+		Name:      asc.ObjectMeta.Name,
+	}
+	return ctrl.Request{NamespacedName: namespacedName}
+}
+
+func newHSCDefault() *aiv1beta1.HypershiftServiceConfig {
+	baseAsc := newASCDefault()
+	return &aiv1beta1.HypershiftServiceConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testName,
+			Namespace: testNamespace,
+		},
+		Spec: aiv1beta1.HypershiftServiceConfigSpec{
+			AgentServiceConfigSpec: aiv1beta1.AgentServiceConfigSpec{
+				FileSystemStorage: baseAsc.Spec.FileSystemStorage,
+				DatabaseStorage:   baseAsc.Spec.DatabaseStorage,
+				ImageStorage:      baseAsc.Spec.ImageStorage,
+			},
+
+			KubeconfigSecretRef: corev1.LocalObjectReference{
+				Name: kubeconfigSecret,
+			},
+		},
+	}
+}
+
+var _ = Describe("HypershiftServiceConfig reconcile", func() {
+	var (
+		ctx      = context.Background()
+		c        client.Client
+		hr       *HypershiftServiceConfigReconciler
+		mockCtrl *gomock.Controller
+	)
+	BeforeEach(func() {
+		c = fakeclient.NewClientBuilder().WithScheme(GetKubeClientSchemes()).Build()
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		hr = &HypershiftServiceConfigReconciler{
+			Client:    c,
+			Scheme:    c.Scheme(),
+			Log:       common.GetTestLog(),
+			Namespace: testNamespace,
+		}
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("Success", func() {
```

Comment:
You don't need a `Context` here you can just use `It` if you don't have any more setup code.

Reply:
This is a place holder for the future tests. We are going to have several test cases here.